hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/HddsDatanodeService.java

```diff
@@ -425,14 +424,15 @@ private DatanodeDetails initializeDatanodeDetails()
     Preconditions.checkNotNull(idFilePath);
     File idFile = new File(idFilePath);
     if (idFile.exists()) {
-      return ContainerUtils.readDatanodeDetailsFrom(idFile);
+      DatanodeDetails details = ContainerUtils.readDatanodeDetailsFrom(idFile);
+      details.setCurrentVersion(DatanodeVersion.CURRENT_VERSION);
```

Comment:
This should already be set in `ContainerUtils#readDatanodeDetailsFrom` [here](https://github.com/apache/ozone/blob/54a5a3c59ce450f81d71d8412fcebbd334d2a59e/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/helpers/DatanodeIdYaml.java#L116-L117)

Reply:
Oh this just kept the current logic DN has, which intentionally overrides the current version already set in the datanode.id yaml file.

I moved this line here because it was placed at a different spot during DN initialization and was confusing to me while I wasted 1/2 hour on debugging why it was already reset to the latest version. So in case anyone debugging this in the future won't waste that amount of time.